blog/2025-08-26-tigris-veo/index.mdx

```diff
@@ -0,0 +1,383 @@
+    }
+
+    console.log(videos);
+
+    const uri = `${videos[0].video?.uri}&key=${API_KEY}`;
+    console.log("Downloading video from:", uri);
+    return uri;
+  } else {
+    throw new Error("No videos generated");
+  }
+}
+```
+
+The above function takes the prompt as an input and calls Veo3 using Google’s
+GenAI SDK generateVideos function. The video generation process takes about a
+minute to complete. That’s why we loop through and check for operation.done
+before extracting the URI. The returned URI represents the download link to the
+video.
+
+Now that we can generate a video, let’s persist it to Tigris and serve it from
+there.
+
+## Set up a Tigris bucket
+
+Tigris is S3‑compatible distributed object storage optimized for low latency.
+Storing videos on Tigris makes them globally available on demand.
+
+1. Create a bucket at https://storage.new (you’ll be prompted to create a Tigris
+   account if needed). Name it, e.g., “veo3-generations”, and click Create.
+2. Create an access key with Read/Write. Toggle “Environment variables” and copy
+   them into your Next.js `.env`.
+3. In “Bucket Rules”, grant Editor permissions.
+
+Add this to `generate/route.ts`:
+
+```typescript
+import { NextRequest, NextResponse } from "next/server";
+import { GoogleGenAI, GeneratedVideo } from "@google/genai";
+import {
+  S3Client,
+  PutObjectCommand,
+  GetObjectCommand,
+} from "@aws-sdk/client-s3";
+import { getSignedUrl } from "@aws-sdk/s3-request-presigner";
+
+const API_KEY = process.env.GOOGLE_API_KEY;
+const ai = new GoogleGenAI({
+  apiKey: API_KEY,
+});
+const VEO3_MODEL_NAME = "veo-3.0-generate-preview";
+
+const S3_BUCKET_NAME = "veo3-generations";
+const S3_REGION = "auto";
+const S3_ENDPOINT = "https://t3.storage.dev";
+const S3_FORCE_PATH_STYLE = false;
+
+const s3 = new S3Client({
+  region: S3_REGION,
+  endpoint: S3_ENDPOINT,
+  forcePathStyle: S3_FORCE_PATH_STYLE,
+  credentials: {
+    accessKeyId: process.env.TIGRIS_STORAGE_ACCESS_KEY_ID!,
+    secretAccessKey: process.env.TIGRIS_STORAGE_SECRET_ACCESS_KEY!,
+  },
+});
+
+async function uploadObject(
+  buffer: Buffer,
+  contentType: string,
+  objectKey: string
+) {
+  await s3.send(
+    new PutObjectCommand({
+      Bucket: S3_BUCKET_NAME,
+      Key: objectKey,
+      Body: buffer,
+      ContentType: contentType,
+    })
+  );
+}
+
+async function getObject(bucket: string, objectKey: string) {
+  const command = new GetObjectCommand({
+    Bucket: bucket,
+    Key: objectKey,
+  });
+  return await getSignedUrl(s3, command, { expiresIn: 60 * 60 });
+}
+```
+
+In the code above, we created an S3 client. Tigris is S3 compatible and supports
+S3 SDK and CLI. We also added two functions (`uploadObject` and `getObject`) to
+upload the videos to our bucket, and then read the video from it.
+
+Use `uploadObject` and `getObject` in `generateVideoFromText`:
+
+```typescript
+async function generateVideoFromText(
+  prompt: string,
+  numberOfVideos = 1,
+): Promise<string[]> {
+  …
```

Comment:
Assumption: this is a continuation of existing code in the function. I find it's better to make sure that the code examples are syntactically valid.

```suggestion
  // ...
```